common/lib/xmodule/xmodule/js/spec/video/completion_spec.js

```diff
@@ -0,0 +1,55 @@
+(function() {
+    'use strict';
+    describe('VideoPlayer completion', function() {
+        var state, oldOTBD;
+
+        beforeEach(function() {
+            oldOTBD = window.onTouchBasedDevice;
+            window.onTouchBasedDevice = jasmine
+                .createSpy('onTouchBasedDevice')
+                .and.returnValue(null);
+
+            state = jasmine.initializePlayer({
+                recordedYoutubeIsAvailable: true,
+                completionEnabled: true,
+                publishCompletionUrl: 'https://example.com/publish_completion_url'
+
+            });
+            state.completionHandler.completeAfterTime = 20;
+        });
+
+        afterEach(function() {
+            $('source').remove();
+            window.onTouchBasedDevice = oldOTBD;
+            state.storage.clear();
+            if (state.videoPlayer) {
+                state.videoPlayer.destroy();
+            }
+        });
+
+        it('calls the completion api when marking an object complete', function() {
+            state.completionHandler.markCompletion(Date.now());
+            expect($.ajax).toHaveBeenCalledWith({
+                url: state.config.publishCompletionUrl,
+                type: 'POST',
+                contentType: 'application/json',
+                dataType: 'json',
+                data: JSON.stringify({completion: 1.0}),
+                success: jasmine.any(Function),
+                error: jasmine.any(Function)
+            });
+        });
+
+        it('calls the completion api on the LMS when the time updates', function() {
+            spyOn(state.completionHandler, 'markCompletion').and.returnValue(null);
+            state.el.trigger('timeupdate', 24.0);
```

Comment:
Consider adding another trigger and checking and changing the check for markCompletion to ensure it was called only once.

Reply:
Done.